spec/tbd_tests_spec.rb

```diff
@@ -2718,4 +2718,100 @@
+    expect(surfaces.size).to eq(180)
+
+    kiva = false
+    surfaces.values.each do |surface|
+      next if kiva
+      kiva = true if surface.has_key?(:kiva)
+    end
+
+    expect(kiva).to be(true)
+    arg = "TotalExposedPerimeter"
+    foundation_kiva_settings = os_model.getFoundationKivaSettings
+    foundation_kiva_settings.setName("TBD-generated Kiva settings template")
+
+    # Generic 1" XPS insulation.
+    xps_25mm = OpenStudio::Model::StandardOpaqueMaterial.new(os_model)
+    xps_25mm.setName("XPS_25mm")
+    xps_25mm.setRoughness("Rough")
+    xps_25mm.setThickness(0.0254)
+    xps_25mm.setConductivity(0.029)
+    xps_25mm.setDensity(28)
+    xps_25mm.setSpecificHeat(1450)
+    xps_25mm.setThermalAbsorptance(0.9)
+    xps_25mm.setSolarAbsorptance(0.7)
+
+    # Typical circa-1980 slab-on-grade (perimeter) insulation setup.
+    kiva_slab = OpenStudio::Model::FoundationKiva.new(os_model)
+    kiva_slab.setName("Kiva slab")
+    kiva_slab.setInteriorHorizontalInsulationMaterial(xps_25mm)
+    kiva_slab.setInteriorHorizontalInsulationWidth(0.6)
+
+    # Basement wall setup (full-height insulation as construction layer)
+    kiva_basement = OpenStudio::Model::FoundationKiva.new(os_model)
+    kiva_basement.setName("Kiva basement")
+
+    # Once XPS and slab/basement objects are generated, assign either one.
+    surfaces.each do |id, surface|
+      next unless surface.has_key?(:kiva)
+      next unless surface.has_key?(:exposed)
+      next unless surface[:exposed] > 0.001
+      next unless surface[:kiva] == :basement || surface[:kiva] == :slab
+
+      found = false
+      os_model.getSurfaces.each do |s|
+        next unless s.nameString == id
+        next unless s.outsideBoundaryCondition.downcase == "foundation"
+
+        found = true
+        exp = surface[:exposed]
+        expect(exp).to be_within(0.01).of(19.20) if id == "g GFloor NWA"
+        expect(exp).to be_within(0.01).of(19.20) if id == "g GFloor NEA"
+        expect(exp).to be_within(0.01).of(19.20) if id == "g GFloor SWA"
+        expect(exp).to be_within(0.01).of(19.20) if id == "g GFloor SEA"
+        expect(exp).to be_within(0.01).of(11.58) if id == "g GFloor S1A"
+        expect(exp).to be_within(0.01).of(11.58) if id == "g GFloor S2A"
+        expect(exp).to be_within(0.01).of(11.58) if id == "g GFloor N1A"
+        expect(exp).to be_within(0.01).of(11.58) if id == "g GFloor N2A"
+        expect(exp).to be_within(0.01).of(3.36) if id == "g Floor C"
+        s.createSurfacePropertyExposedFoundationPerimeter(arg, surface[:exposed])
+        s.setAdjacentFoundation(kiva_basement) if surface[:kiva] == :basement
+        s.setAdjacentFoundation(kiva_slab) if surface[:kiva] == :slab
+      end
+
+      # Loop through basement wall surfaces and assign foundation object.
+      surfaces.each do |i, surf|
+        next unless found
+        next unless surf.has_key?(:foundation)
+        next unless id == surf[:foundation]
```

Comment:
I'm a bit confused about what is going on here.  Basement walls are not tagged with `surface[:kiva] = :basement` but instead the walls reference the basement floor as the `surface[:foundation]`?  This is potentially more confusing because of confusion between tbd foundation object and the kiva foundation object.  Might help to add some comments

Reply:
Sure, I could add more comments. TBD surfaces tagged with _:kiva_ are strictly floors with "Foundation" as outside boundary condition (with _:basement_ or _:slab_ as 2x possible values). TBD surfaces with _:foundation_ tags are strictly walls that are part of the same Kiva foundation as the adjacent floor. The _:foundation_ tag in fact holds the name of the foundation floor. It's just handy to quickly identify which walls belong to which foundation/floor.

Reply:
The most recent commit brought in some sorely needed clarity, so most of the previous comments here no longer apply.